### PR TITLE
Capture-able Zones

### LINKED ISSUE #78: Capture-able Zones
**Describe the solution you'd like**
Zones available for the player to capture should be implemented.
Player zones should use a trigger when entered/exited to update their status.

### PR DESCRIPTION
**Pull request by AlexDyre**
## Description
Implements capture-able zones to the game level.
Zones captured by a faction provide addition resource income.
A zone is captured by a faction once only units for a single faction exist inside the zone.
The display of zones can be toggled via the "ZoneDisplay" key (default: Z).

## Related Issue
Closes #78 

## Motivation and Context
As a player, I should be able to capture zones from the enemy, to provide more resources to my faction.

## How Has This Been Tested?
All existing tests pass.
Game functions as expected with the addition changes.

## Screenshots (when applicable):
image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Code Refactor (styling fix, extracting methods, etc. Does not cause any functionality changes)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: DEFCON Z/Assets/Scripts/GameLevel/Zone.cs [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+
+        /// <summary>
+        /// Updates the current zone
+        /// Itterates through units inside zone and calculates the correct owner of the zone
+        /// </summary>
+        private void UpdateZone()
+        {
+            Faction _owner = null;
+            bool _multipleFactions = false;
+            int _unitCount = 0;
```
Comment (ferdinh) [inline DEFCON Z/Assets/Scripts/GameLevel/Zone.cs]: You can use `unitsInZone.Count`.

File: DEFCON Z/Assets/Scripts/GameLevel/Zone.cs [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+                    zoneOwner.Resource.MaxResourcePoint -= zoneResourceValue;
+                }
+
+                // Assign the new zone owner
+                zoneOwner = faction;
+                // Add the resource cap boost to the new owner
+                zoneOwner.Resource.MaxResourcePoint += zoneResourceValue;
+
+                // Update the display color of the zone
+                if (faction == null)
```
[the remark this answers is not in the document]
Comment (AlexDyre) [reply DEFCON Z/Assets/Scripts/GameLevel/Zone.cs]: Not a bad suggestion!

File: DEFCON Z/Assets/Scripts/GameLevel/Zone.cs [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+                if (_owner == null)
+                {
+                    _owner = _unit.FactionOwner;
+                }
+                else
+                {
+                    // Check if the current calculated owner is different to the unit
+                    if (_unit.FactionOwner != _owner)
+                    {
+                        _multipleFactions = true;
```
[the remark this answers is not in the document]
Comment (AlexDyre) [reply DEFCON Z/Assets/Scripts/GameLevel/Zone.cs]: Done!

**Comment by ferdinh** [comment]
Nice work @AlexDyre.

**Pull request merged**